packages/block-editor/src/hooks/border-width.js

```diff
@@ -43,16 +54,34 @@ export const BorderWidthEdit = ( props ) => {
 			},
 		};
 
+		const hasZeroWidth = parseFloat( newWidth ) === 0;
+
+		// Setting the border width explicitly to zero will also set the
+		// border style prop to `none`. This style will only be applied if
+		// border style support has also been opted into.
+		if ( hasZeroWidth ) {
+			newStyle.border.style = 'none';
```

Comment:
This is testing well 👍 

One thing I noticed: with the style removed, the inline styles on the frontend of my group block appear so:

`border-color:#f3ff98;border-radius:22px;border-style:none;border-width:0px;color:#1d1dff`

We're caching the previous border styles, so does it make sense to also remove `border-color` and associated class name `has-border-color`? 

I'm only thinking that, if the post is saved with zero width, the inline style and class for the border color remain unnecessarily forever.

Reply:
Thanks, I wasn't sure about how this might be used. I defaulted to changing as little as possible on the border block supports.

It could be possible to use the remaining border properties on child elements via inheriting them through CSS. I don't believe there is the ability to conditionally apply styles in block supports/global styles i.e. only apply the border color if border width > 0. That would mean I'll have to make a similar change as for style and unset the other border support options.

Perhaps I was overthinking things. I agree that it would be neater to remove the color, radius styles/classes etc if they aren't being used.

With the future refinements to the sidebar controls, in particular, the progressive discovery system, it might be that an entire border "instance" is added or removed. Which would effectively reset all border support properties in one go. I'm not sure at this point how that would play with theme set borders although I assume by the time that is implemented we'll have access to the theme set values for block contexts with the block editor, similar to what's available via the state in the full site editor.

In the end, this PR was to satisfy the [requested border support UI improvements](https://github.com/WordPress/gutenberg/issues/31337), and allow the user to override a theme set border and disable it. 

I'll update the PR to reset the other border properties. Thanks again for the thorough review 👍